packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -2132,6 +2132,15 @@ function beginWork(
           resetHydrationState();
           break;
         case HostComponent:
+          if (
+            workInProgress.mode & ConcurrentMode &&
+            renderExpirationTime !== Never &&
+            shouldDeprioritizeSubtree(workInProgress.type, newProps)
+          ) {
+            // Schedule this fiber to re-render at offscreen priority. Then bailout.
+            workInProgress.expirationTime = workInProgress.childExpirationTime = Never;
+            return null;
+          }
           pushHostContext(workInProgress);
```

Comment:
Oh wait, I think this context push needs to go above the return

Reply:
fixd